gprimitives/src/lib.rs

```diff
@@ -216,3 +227,96 @@ impl FromStr for CodeId {
 pub struct ReservationId([u8; 32]);
 
 macros::impl_primitive!(new zero into_bytes from_u64 from_h256 display debug, ReservationId);
+
+#[cfg(test)]
+mod tests {
+    extern crate alloc;
+
+    use crate::ActorId;
+    use alloc::format;
+    use core::str::FromStr;
+
+    fn actor_id() -> ActorId {
+        ActorId::from_str("0x6a519a19ffdfd8f45c310b44aecf156b080c713bf841a8cb695b0ea5f765ed3e")
+            .unwrap()
+    }
+
+    #[test]
+    #[should_panic]
+    fn wrong_format1_test() {
```

Comment:
names are unintuitive, please describe test cases

Reply:
moreover, why do we return err on this cases instead of supporting it?
if it's temporary, please open an issue

at least case of +.2 looks valid for me